src/com/etendoerp/dependencymanager/process/SelectorChangeVersion.java

```diff
@@ -88,6 +100,62 @@ protected JSONObject execute(Map<String, Object> parameters, String content) {
     return jsonResponse;
   }
 
+  /**
+   * Processes core dependency information for the specified package version and returns the results as a JSON object.
+   *
+   * <p>This method checks if a core dependency exists in the package version's dependency list with an artifact
+   * matching the core identifier {@code ETENDO_CORE}. If found, it retrieves the core version range. If no core dependency
+   * is found or if the version is empty, it uses the {@code fromCore} and {@code latestCore} values in the package version
+   * to determine the version range. The method then populates a JSON object with the core version range and
+   * the recommended core update version.</p>
+   *
+   * @param pkgVersion the {@code PackageVersion} object representing the package version containing dependency information
+   * @return a {@code JSONObject} with core dependency details, including:
+   *         <ul>
+   *           <li>{@code CORE_VERSION_RANGE}: The version range for the core dependency if available, otherwise "No version range available".</li>
+   *           <li>{@code UPDATE_TO_CORE_VERSION}: The recommended core update version range if available.</li>
+   *           <li>In case of an error, includes an "error" field with a description of the issue.</li>
+   *         </ul>
+   */
+  private JSONObject processCoreDependency(PackageVersion pkgVersion) {
+    JSONObject result = new JSONObject();
+    try {
+      PackageDependency coreDep = pkgVersion.getETDEPPackageDependencyList().stream()
+              .filter(dep -> StringUtils.equals(ETENDO_CORE, dep.getArtifact()))
+              .findFirst()
+              .orElse(null);
+
+      String coreVersionRange;
+      String updateToCoreVersion;
+
+      if (coreDep == null || StringUtils.isEmpty(coreDep.getVersion())) {
```

Comment:
Version can be null? Maybe isBlank is a better validation

Reply:
Solved!